daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -396,4 +408,22 @@ object Engine {
     InitialSeeding.TransactionSeed(
       crypto.Hash.deriveTransactionSeed(submissionSeed, participant, submissionTime))
 
+  private def profileName(tx: Tx.Transaction, meta: Tx.Metadata): String = {
+    val hash = meta.nodeSeeds(0)._2.toHexString
+    val desc =
+      if (tx.roots.length == 1) {
+        val makeDesc = (kind: String, tmpl: Ref.Identifier, extra: Option[String]) =>
+          s"${kind}:${tmpl.qualifiedName.name}${extra.map(extra => s":${extra}").getOrElse("")}"
+        tx.nodes.get(tx.roots(0)).toList.head match {
+          case create: NodeCreate[_, _] => makeDesc("create", create.coinst.template, None)
+          case exercise: NodeExercises[_, _, _] =>
+            makeDesc("exercise", exercise.templateId, Some(exercise.choiceId.toString))
+          case fetch: NodeFetch[_, _] => makeDesc("fetch", fetch.templateId, None)
+          case lookup: NodeLookupByKey[_, _] => makeDesc("lookup", lookup.templateId, None)
+        }
+      } else {
+        s"compound:${tx.roots.length}"
+      }
+    s"${meta.submissionTime}-${desc}-${hash}.json"
```

Comment:
It does not work for `LF-REPL` as the `submissionTime` is always EPOCH and the sequence of `hashes` may repeated (for a same program /scenario they will repeat).
The fix time and fix sequence of seed is done to have reproducible runs. 
In case of sandbox it should be fine.

Reply:
`LF-REPL` does not use the engine but rather the speedy machine directly. for `daml-lf-repl profile` you need to specify an output file directly hence.